data/statuses.js

```diff
@@ -728,8 +729,23 @@ let BattleStatuses = {
 			// TODO work on display for HP
 			this.add('-heal', pokemon, pokemon.getHealth, '[silent]');
 		},
+		onSwitchIn(pokemon) { // Putting Eternamax in onSwitchIn so it shows up eveytime Eternatus switches in.
+			if (pokemon.species !== 'Eternatus-Eternamax') return; // Special for Eternatus' Eternamax forme
+			pokemon.removeVolatile('substitute');
+			this.add('-start', pokemon, 'Eternamax');
+			this.effectData.duration = 0;
+			pokemon.maxhp = Math.floor(pokemon.maxhp * 2);
+			pokemon.hp = Math.floor(pokemon.hp * 2);
+			this.add('-heal', pokemon, pokemon.getHealth, '[silent]');
+		},
 		onFlinch: false,
 		onBeforeSwitchOut(pokemon) {
+			if (pokemon.species === 'Eternatus-Eternamax') {
+				pokemon.maxhp = Math.floor(pokemon.maxhp / 2); // TODO prevent maxhp loss
+				pokemon.hp = Math.floor(pokemon.hp / 2);
+				if (pokemon.hp <= 0) pokemon.hp = 1;
+				return;
+			}
```

Comment:
Why not just do `if (pokemon.species === 'Eternatus-Eternamax') return;` right before `this.hint("Dynamax ended.");`?